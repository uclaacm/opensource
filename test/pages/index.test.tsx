import React from 'react';
import Home from '../../pages/index';
import { render } from '../testUtils';

describe('Home page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Home numRepos={0} recentEvents={[]} projNumToDisplay={0} />, {});
    expect(asFragment()).toMatchSnapshot();
  });
});
