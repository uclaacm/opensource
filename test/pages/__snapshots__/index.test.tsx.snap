// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar"
  >
    <div
      class="navbar-brand"
    >
      <a
        href="/"
      >
        <img
          alt="ACM at UCLA Logo"
          src="https://design.uclaacm.com/assets/logos/acm-logo-wordmark-extended.png"
        />
      </a>
    </div>
    <div
      class="navbar-items"
    >
      <div
        class="navbar-link"
      >
        <a
          href="/projects"
        >
          projects
        </a>
        <a
          href="/contribute"
        >
          contribute
        </a>
        <button>
          GitHub
        </button>
      </div>
    </div>
  </nav>
  <main>
    <div
      class="container"
    >
      <h1>
        open source at 
        <a
          href="https://uclaacm.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          ACM at UCLA
        </a>
      </h1>
      <hr />
      <p
        class="knockout-description"
      >
        at the largest computer science community at UCLA,
        <br />
         
        <b>
          we care about open-source.
        </b>
      </p>
      <div
        class="row"
      >
        <div
          class="col-6"
        >
          <div
            class="card"
          >
            <div
              class="card-body"
            >
              <a
                href="/projects"
              >
                <h3>
                  Projects →
                </h3>
                <p>
                  All of our projects are open-source! Literally, all of them.
                </p>
              </a>
            </div>
          </div>
        </div>
        <div
          class="col-6"
        >
          <div
            class="card"
          >
            <div
              class="card-body"
            >
              <a
                href="https://github.com/uclaacm/"
                rel="noopener noreferrer"
                target="_blank"
              >
                <h3>
                  Learning →
                </h3>
                <p>
                  Pick up the skills to write great software.
                </p>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="row mt-2"
      >
        <div
          class="col-6"
        >
          <div
            class="card"
          >
            <div
              class="card-body"
            >
              <a
                href="https://github.com/uclaacm/"
                rel="noopener noreferrer"
                target="_blank"
              >
                <h3>
                  Events →
                </h3>
                <p>
                  We love talking about open-source. Come talk with us!
                </p>
              </a>
            </div>
          </div>
        </div>
        <div
          class="col-6"
        >
          <div
            class="card"
          >
            <div
              class="card-body"
            >
              <a
                href="/contribute"
              >
                <h3>
                  Contribute →
                </h3>
                <p>
                  Make your first pull request with us!
                </p>
              </a>
            </div>
          </div>
        </div>
      </div>
      <hr
        class="mt-2"
      />
      <h2>
        featured project
      </h2>
      <div
        class="card"
      >
        <div
          class="row"
        >
          <div
            class="col-6"
          >
            <a
              href="https://bufferbuffet.uclaacm.com/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <div
                style="display: block; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
              >
                <div
                  style="display: block; box-sizing: border-box; padding-top: 80%;"
                />
                <img
                  alt="buffer buffet landing splash"
                  decoding="async"
                  sizes="100vw"
                  src="/_next/image?url=%2Fprojects%2Fbuffer-buffet.png&w=3840&q=75"
                  srcset="/_next/image?url=%2Fprojects%2Fbuffer-buffet.png&w=640&q=75 640w, /_next/image?url=%2Fprojects%2Fbuffer-buffet.png&w=750&q=75 750w, /_next/image?url=%2Fprojects%2Fbuffer-buffet.png&w=828&q=75 828w, /_next/image?url=%2Fprojects%2Fbuffer-buffet.png&w=1080&q=75 1080w, /_next/image?url=%2Fprojects%2Fbuffer-buffet.png&w=1200&q=75 1200w, /_next/image?url=%2Fprojects%2Fbuffer-buffet.png&w=1920&q=75 1920w, /_next/image?url=%2Fprojects%2Fbuffer-buffet.png&w=2048&q=75 2048w, /_next/image?url=%2Fprojects%2Fbuffer-buffet.png&w=3840&q=75 3840w"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
              </div>
            </a>
          </div>
          <div
            class="col-6"
          >
            <div
              class="card-body"
            >
              <h3
                class="mt-1"
              >
                <a
                  href="https://bufferbuffet.uclaacm.com/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Buffer Buffet
                </a>
              </h3>
              <p>
                <span
                  class="dev-language-badge lang-js"
                />
                 js
                <span>
                   • react, x86
                </span>
              </p>
              <p>
                All of our projects are open-source! Literally, all of them.
              </p>
              <a
                href="https://github.com/uclaacm/buffer-buffet"
                rel="noopener noreferrer"
                target="_blank"
              >
                GitHub Repository
              </a>
            </div>
          </div>
        </div>
      </div>
      <h2>
        what we've been doing recently...
      </h2>
      <p>
        repositories: 0
      </p>
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <p>
            see more activity 
            <a
              href="https://github.com/uclaacm/"
              rel="noopener noreferrer"
              target="_blank"
            >
              on our org
            </a>
            !
          </p>
        </div>
      </div>
    </div>
  </main>
  <footer>
    <div
      class="container"
    >
      made by 
      <a
        href="https://uclaacm.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        ACM at UCLA
      </a>
      , with 
      <a
        href="https://nextjs.org/docs"
        rel="noopener noreferrer"
        target="_blank"
      >
        Next.js
      </a>
    </div>
  </footer>
</DocumentFragment>
`;
